tests/e2e/network/ensureloadbalancer.go

```diff
@@ -87,6 +87,46 @@ var _ = Describe("Ensure LoadBalancer", func() {
 		tc = nil
 	})
 
+	It("should support BYO public IP", func() {
+		By("creating a public IP with tags")
+		ipName := basename + "-public-IP" + string(uuid.NewUUID())[0:4]
+		pip := defaultPublicIPAddress(ipName)
+		expectedTags := map[string]*string{
+			"foo": to.StringPtr("bar"),
+		}
+		pip.Tags = expectedTags
+		pip, err := utils.WaitCreatePIP(tc, ipName, tc.GetResourceGroup(), pip)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pip.Tags).To(Equal(expectedTags))
+		targetIP := to.String(pip.IPAddress)
+		utils.Logf("created pip with address %s", targetIP)
+
+		defer func() {
+			By("cleaning up")
+			err = utils.DeletePIPWithRetry(tc, ipName, "")
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		By("creating a service referencing the public IP")
+		service := utils.CreateLoadBalancerServiceManifest(testServiceName, nil, labels, ns.Name, ports)
+		service = updateServiceBalanceIP(service, false, targetIP)
+		_, err = cs.CoreV1().Services(ns.Name).Create(context.TODO(), service, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		err = utils.WaitUpdateServiceExposure(cs, ns.Name, testServiceName, targetIP, true)
+		Expect(err).NotTo(HaveOccurred())
+
```

Comment:
also check the connectivity of service PIP here?

Reply:
done